core/domain/feedback_jobs_one_off_test.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Feedback-related jobs."""
+
+import ast
+
+from core.domain import feedback_jobs_one_off
+from core.platform import models
+from core.tests import test_utils
+
+(feedback_models,) = models.Registry.import_models([models.NAMES.feedback])
+taskqueue_services = models.Registry.import_taskqueue_services()
+
+
+class GeneralFeedbackThreadUserOneOffJobTest(test_utils.GenericTestBase):
+    """Tests for ExpSummary aggregations."""
+
+    ONE_OFF_JOB_MANAGERS_FOR_TESTS = [
+        feedback_jobs_one_off.GeneralFeedbackThreadUserOneOffJob]
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            feedback_jobs_one_off.GeneralFeedbackThreadUserOneOffJob
+            .create_new())
+        feedback_jobs_one_off.GeneralFeedbackThreadUserOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+        stringified_output = (
+            feedback_jobs_one_off.GeneralFeedbackThreadUserOneOffJob
+            .get_output(job_id))
+
+        eval_output = [ast.literal_eval(stringified_item)
+                       for stringified_item in stringified_output]
+        output = [(eval_item[0], int(eval_item[1]))
+                  for eval_item in eval_output]
+        return output
+
+    def test_successful_migration(self):
```

Comment:
Add a test to ensure that the job does nothing for stuff that already has populated fields.

Reply:
Done.

Reply:
Sorry I think I was unclear. I meant for a model that already has user_id and thread_id fields populated.

Reply:
I have added this into test_multiple_feedbacks, but I can probably also add it as a separate test.

Reply:
Yes, separate test sounds good, to be clearer about the purpose.

Reply:
Done.